src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -121,6 +119,17 @@ public static SegmentInfos readSegmentInfos(Directory directory) throws IOExcept
     }
 
     /**
+     * Returns an iterable that allows to iterate over all files in this segments info
+     */
+    public static Iterable<String> files(SegmentInfos infos) throws IOException {
+        final List<Collection<String>> list = new ArrayList<>();
+        for (SegmentCommitInfo info : infos) {
```

Comment:
Should this include the segments file? (e.g. call infos.files() instead) The old listAll() would have included it too.